From a5101c6cf190774a05fb933300b8628df5af0849 Mon Sep 17 00:00:00 2001
From: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
Date: Thu, 23 May 2019 23:56:50 -0700
Subject: [PATCH] FROMGIT: ASoC: SOF: hda: set autosuspend delay for hda bus device

BugLink: https://bugs.launchpad.net/bugs/1826181

Set the autosuspend delay to 2s for hda bus device.
This is needed to prevent the codec driver from entering
runtime PM during stress tests involving back-to-back
stream open/close.

The legacy driver forbids autosuspend for some devices
where a non-zero autosuspend delay causes pops/clicks.
This needs to be addressed in the SOF driver when those
devices are supported.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
(cherry picked from commit a20b3386f7352952fab976c96a723c2e971d3501
git://github.com/thesofproject/linux.git)
Signed-off-by: Hui Wang <hui.wang@canonical.com>
---
 sound/soc/sof/intel/hda-codec.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sound/soc/sof/intel/hda-codec.c b/sound/soc/sof/intel/hda-codec.c
index c711792534da..edac308fcf9e 100644
--- a/sound/soc/sof/intel/hda-codec.c
+++ b/sound/soc/sof/intel/hda-codec.c
@@ -138,6 +138,7 @@ static int hda_codec_probe(struct snd_sof_dev *sdev, int address)
 int hda_codec_probe_bus(struct snd_sof_dev *sdev)
 {
 	struct hdac_bus *bus = sof_to_bus(sdev);
+	struct hda_bus *hbus = sof_to_hbus(sdev);
 	int i, ret;
 
 	/* probe codecs in avail slots */
@@ -154,6 +155,9 @@ int hda_codec_probe_bus(struct snd_sof_dev *sdev)
 		}
 	}
 
+	/* set autosuspend delay for hda bus device */
+	snd_hda_set_power_save(hbus, SND_SOF_SUSPEND_DELAY_MS);
+
 	return 0;
 }
 EXPORT_SYMBOL(hda_codec_probe_bus);
-- 
2.17.1

